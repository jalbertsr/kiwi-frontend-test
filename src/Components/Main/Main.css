h3 {
  line-height: 0;
  margin-top: 1%;
  display: inline-block;
}

.pagination {
  position: absolute;
  right: 0;
}

.spacePagination {
  margin-bottom: 3%;
}