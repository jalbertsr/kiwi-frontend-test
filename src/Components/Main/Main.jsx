import React, { Component } from 'react'
import { Row, Col, Pagination } from 'antd'
import uuidv4 from 'uuid/v4'
import moment from 'moment'

import styles from './Main.css'
import Service from '../../Services/flightService'
import FlightCard from '../FlightCard/FlightCard'
import SearchForm from '../SearchForm/SearchForm'

export default class Main extends Component {
  constructor () {
    super()
    this.state = {
      from: 'Barcelona',
      to: 'Brno',
      departure: moment().format('DD/MM/YYYY'),
      returning: moment().add(5, 'days').format('DD/MM/YYYY'),
      loading: true,
      results: [],
      page: 1,
      resultsLength: 0,
      slice: {
        init: 0,
        final: 18
      }
    }
  }

  handleSubmit = (e) => {
    e.preventDefault()
    this.getResults(this.state.from, this.state.to, 
      this.state.departure, this.state.returning, {init: 0, final: 18})
  }

  handleDates = (departure, returning) => {
    this.setState({departure, returning})
  }

  handleChange = (value) => {
    this.setState(value)
  }

  handleChangePage = (page) => {
    let init, final
    const coeficient = Math.abs(page - this.state.page) * 18
    if (page > this.state.page) {
      init = this.state.slice.init + coeficient
      final = this.state.slice.final + coeficient
    } else if (page < this.state.page) {
      init = this.state.slice.init - coeficient
      final = this.state.slice.final - coeficient
    }
    this.setState({
      page: page,
      slice: { init, final }
    })
    this.getResults(this.state.from, this.state.to, 
      this.state.departure, this.state.returning, this.state.slice)
  }

  getResults = (_from, to, departure, returning, slice) => {
    this.setState({ loading: true })
    Service.getFlights(_from, to, departure, returning)
      .then(info => {
        this.setState({
          loading: false,
          resultsLength: info.data.length,
          results: info.data.slice(slice.init, slice.final)
        })
      })
  }

  componentDidMount () {
    this.getResults(this.state.from, this.state.to, 
      this.state.departure, this.state.returning, {init: 0, final: 18})
  }

  render () {
    return (
      <div>
        <SearchForm
          handleSubmit={this.handleSubmit}
          handleChange={this.handleChange}
          handleDates={this.handleDates}
        />
        <Row>
          <Col className={styles.spacePagination} sm={19} offset={2}>
            <Pagination
              onChange={this.handleChangePage}
              className={styles.pagination}
              current={this.state.page}
              defaultCurrent={1}
              total={this.state.resultsLength}
            />
            <h3>Total Results: {this.state.resultsLength}</h3>
          </Col>
        </Row>
        <Row gutter={14}>
          {this.state.results.map((flight, i) =>
            <Col offset={2} sm={8} md={5} key={uuidv4()}>
              <FlightCard
                loading={this.state.loading}
                title={`From ${flight.cityFrom} to ${flight.cityTo}`}
                description={flight}
                departure={this.state.departure}
                returning={this.state.returning}
                index={i}
              />
            </Col>
          )}
        </Row>
      </div>
    )
  }
}
