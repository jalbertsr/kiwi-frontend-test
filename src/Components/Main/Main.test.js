import Main from './Main'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

configure({ adapter: new Adapter() })

/* eslint-disable no-unused-vars */
describe('Main', () => {
  test('renders correctly', () => {
    const component = shallow(<Main />)
    expect(component).toMatchSnapshot()
  })
})
