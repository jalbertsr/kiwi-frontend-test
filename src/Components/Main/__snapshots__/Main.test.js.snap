// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SearchForm
          handleChange={[Function]}
          handleDates={[Function]}
          handleSubmit={[Function]}
        />,
        <Row
          gutter={0}
        >
          <Col
            className="spacePagination"
            offset={2}
            sm={19}
          >
            <_a
              className="pagination"
              current={1}
              defaultCurrent={1}
              onChange={[Function]}
              prefixCls="ant-pagination"
              selectPrefixCls="ant-select"
              total={0}
            />
            <h3>
              Total Results: 
              0
            </h3>
          </Col>
        </Row>,
        <Row
          gutter={14}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleChange": [Function],
          "handleDates": [Function],
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Col
            className="spacePagination"
            offset={2}
            sm={19}
          >
            <_a
              className="pagination"
              current={1}
              defaultCurrent={1}
              onChange={[Function]}
              prefixCls="ant-pagination"
              selectPrefixCls="ant-select"
              total={0}
            />
            <h3>
              Total Results: 
              0
            </h3>
          </Col>,
          "gutter": 0,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <_a
                className="pagination"
                current={1}
                defaultCurrent={1}
                onChange={[Function]}
                prefixCls="ant-pagination"
                selectPrefixCls="ant-select"
                total={0}
              />,
              <h3>
                Total Results: 
                0
              </h3>,
            ],
            "className": "spacePagination",
            "offset": 2,
            "sm": 19,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "pagination",
                "current": 1,
                "defaultCurrent": 1,
                "onChange": [Function],
                "prefixCls": "ant-pagination",
                "selectPrefixCls": "ant-select",
                "total": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Total Results: ",
                  0,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Total Results: ",
                0,
              ],
              "type": "h3",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [],
          "gutter": 14,
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SearchForm
            handleChange={[Function]}
            handleDates={[Function]}
            handleSubmit={[Function]}
          />,
          <Row
            gutter={0}
          >
            <Col
              className="spacePagination"
              offset={2}
              sm={19}
            >
              <_a
                className="pagination"
                current={1}
                defaultCurrent={1}
                onChange={[Function]}
                prefixCls="ant-pagination"
                selectPrefixCls="ant-select"
                total={0}
              />
              <h3>
                Total Results: 
                0
              </h3>
            </Col>
          </Row>,
          <Row
            gutter={14}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleChange": [Function],
            "handleDates": [Function],
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Col
              className="spacePagination"
              offset={2}
              sm={19}
            >
              <_a
                className="pagination"
                current={1}
                defaultCurrent={1}
                onChange={[Function]}
                prefixCls="ant-pagination"
                selectPrefixCls="ant-select"
                total={0}
              />
              <h3>
                Total Results: 
                0
              </h3>
            </Col>,
            "gutter": 0,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <_a
                  className="pagination"
                  current={1}
                  defaultCurrent={1}
                  onChange={[Function]}
                  prefixCls="ant-pagination"
                  selectPrefixCls="ant-select"
                  total={0}
                />,
                <h3>
                  Total Results: 
                  0
                </h3>,
              ],
              "className": "spacePagination",
              "offset": 2,
              "sm": 19,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "pagination",
                  "current": 1,
                  "defaultCurrent": 1,
                  "onChange": [Function],
                  "prefixCls": "ant-pagination",
                  "selectPrefixCls": "ant-select",
                  "total": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Total Results: ",
                    0,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Total Results: ",
                  0,
                ],
                "type": "h3",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [],
            "gutter": 14,
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
